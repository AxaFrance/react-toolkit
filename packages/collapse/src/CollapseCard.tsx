import React from 'react';

import { Constants } from '@axa-fr/react-toolkit-core';
import CollapseCardBase from './CollapseCardBase';
import Body from './Body';
import Header from './Header';

const defaultProps = {
  ...Constants.defaultProps,
  defaultCollapse: true,
  index: 0,
};

type Props = Partial<typeof defaultProps> & {
  children?: React.ReactNode[];
  title?: string;
  id?: string;
};

const CollapseCard = ({
  title,
  className,
  id,
  classModifier,
  defaultCollapse,
  children,
  index,
}: Props) => {
  const [collapse, setCollapse] = React.useState(defaultCollapse);

  return (
    <CollapseCardBase
      collapse={collapse}
      onToggle={() => setCollapse(!collapse)}
      title={title}
      className={className}
      id={id}
      classModifier={classModifier}
      index={index}>
      {children}
    </CollapseCardBase>
  );
};

CollapseCard.defaultProps = defaultProps;

CollapseCard.Body = Body;
CollapseCard.Header = Header;

export default CollapseCard;
