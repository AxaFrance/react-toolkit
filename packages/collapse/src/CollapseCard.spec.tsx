import React, { ComponentProps } from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import CollapseCard from './CollapseCard';

const setup = (
  props: Omit<ComponentProps<typeof CollapseCard>, 'children'>
) => {
  const utils = render(
    <CollapseCard {...props}>
      <CollapseCard.Header>Collapsible Group Item #1</CollapseCard.Header>
      <CollapseCard.Body>
        Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus
        terry richardson ad squid. 3 wolf moon officia aute, non cupidatat
        skateboard dolor brunch. Food truck quinoa nesciunt laborum eiusmod.
        Brunch 3 wolf moon tempor, sunt aliqua put a bird on it squid
        single-origin coffee nulla assumenda shoreditch et. Nihil anim keffiyeh
        helvetica, craft beer labore wes anderson cred nesciunt sapiente ea
        proident. Ad vegan excepteur butcher vice lomo. Leggings occaecat craft
        beer farm-to-table, raw denim aesthetic synth nesciunt you probably
        haven&apos;t heard of them accusamus labore sustainable VHS.
      </CollapseCard.Body>
    </CollapseCard>
  );
  return { ...utils };
};

describe('CollapseCard', () => {
  it('renders CollapseCard correctly', () => {
    const { asFragment } = setup({ id: 'collapse1' });
    expect(asFragment()).toMatchSnapshot();
  });

  it('onToggle call onToggle', async () => {
    const onToggle = jest.fn();
    const { getByRole } = setup({
      onToggle,
      id: 'collapse1',
      classModifier: 'classModifier',
      className: '',
      isOpen: false,
    });
    const header = getByRole('link');

    fireEvent.click(header);

    await waitFor(() => {
      expect(onToggle).toHaveBeenCalledWith({
        isOpen: true,
        id: 'collapse1',
        index: 0,
      });
    });
  });

  it('should render any child', () => {
    const { container } = render(
      <CollapseCard>
        <h1>h1</h1>
        <p>p</p>
      </CollapseCard>
    );
    expect(container.querySelector('h1')).toBeInTheDocument();
    expect(container.querySelector('p')).toBeInTheDocument();
  });
});
