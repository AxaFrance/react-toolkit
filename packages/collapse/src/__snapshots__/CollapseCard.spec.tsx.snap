// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseCard renders CollapseCard correctly 1`] = `
<DocumentFragment>
  <div
    class="af-accordion__item"
  >
    <div
      class="af-accordion__item-header"
      id="collapse1"
      role="tab"
    >
      <a
        aria-controls="collapse1body"
        aria-expanded="false"
        class="af-accordion__item-toggle"
        data-toggle="isOpen"
        href="#collapse1body"
      >
        <h3
          class="af-accordion__item-title"
        >
          <span
            class="glyphicon glyphicon-menu-down"
          />
          Collapsible Group Item #1
        </h3>
      </a>
    </div>
    <div
      aria-labelledby="collapse1"
      class="af-accordion__collapse af-accordion__collapse--close"
      id="collapse1body"
      role="tabpanel"
      style="max-height: 0px;"
    >
      <div
        class="af-accordion__block"
      >
        Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. 3 wolf moon officia aute, non cupidatat skateboard dolor brunch. Food truck quinoa nesciunt laborum eiusmod. Brunch 3 wolf moon tempor, sunt aliqua put a bird on it squid single-origin coffee nulla assumenda shoreditch et. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident. Ad vegan excepteur butcher vice lomo. Leggings occaecat craft beer farm-to-table, raw denim aesthetic synth nesciunt you probably haven't heard of them accusamus labore sustainable VHS.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
